### PR TITLE
Added python import-profiler

### PR DESCRIPTION
**Pull request by 3eka**
Required by bzr (at least...)

### CODE REVIEW

File: components/python/import-profiler/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2017 Predrag Zečević.  All rights reserved.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+PATH=/usr/bin:/usr/gnu/bin:/usr/sbin
```
Comment (alarcher) [inline components/python/import-profiler/Makefile]: This line can be removed.

**Comment by alarcher** [comment]
@pyhalov @xen0l Any opinion about Python 2.7 versus Python 3.4?

**Comment by pyhalov** [comment]
Why can't we build it for both versions?

File: components/python/import-profiler/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+COMPONENT_POST_INSTALL_ACTION= \
+  gfind $(PROTO_DIR)$(PYTHON_VENDOR_PACKAGES.32) -type f \
+    -exec gawk '/^\#!.*python/{print FILENAME}{nextfile}' {} + \
+  | xargs gsed -i -e '1 s;^.*$$;\#!$(PYTHON.32);'
+
+# setp.py needs README.md, but file is missing in archive?
+COMPONENT_PRE_BUILD_ACTION=cd $(SOURCE_DIR) && echo "Import profiler to find bottlenecks in import times." > README.md
+
+# common targets
+build:		$(BUILD_32_and_64)
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: As component  doesn't provide binary components you can do just 32-bit build

**Comment by 3eka** [comment]
Hi @pyhalov,

I have tried to make suggested changed, but I get errors when trying to compile:
```
/usr/bin/pkgmogrify -D PYV=### /export/home/zeka/oi-userland/components/python/import-profiler/build/mkgeneric-python \
        /export/home/zeka/oi-userland/transforms/mkgeneric import-profiler-PYVER.p5m > /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.p5m
if [ -f import-profiler-GENFRAG.p5m ]; then cat import-profiler-GENFRAG.p5m >> /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.p5m; fi
/usr/bin/pkgmogrify -D PERL_ARCH=i86pc-solaris-64int -D PERL_VERSION=5.22 -D PG_VERSION=9.4 -D PG_VERNUM=94 -D PG_BASEPKG=database/postgres-94 -D MYSQL_VERSION=10.1 -D MYSQL_VERNUM=101 -D MYSQL_BASEPKG=database/mariadb-101 -D MACH=i386 -D MACH32=i86 -D MACH64=amd64 -D PUBLISHER=userland -D PUBLISHER_LOCALIZABLE=userland-localizable -D CONSOLIDATION=userland -D BUILD_VERSION=5.11-2017.0.0.0 -D SOLARIS_VERSION=2.11 -D OS_VERSION=5.11 -D PKG_SOLARIS_VERSION=5.11 -D HUMAN_VERSION= -D IPS_COMPONENT_VERSION=0.0.3 -D COMPONENT_VERSION=0.0.3 -D COMPONENT_PROJECT_URL=https://pypi.python.org/pypi/import-profiler -D COMPONENT_ARCHIVE_URL=https://pypi.python.org/packages/1f/cf/6a60ab41279a8c9a75c6370ccd41449bcc09a39385dd6c282bd8c123053a/import_profiler-0.0.3.tar.gz -D COMPONENT_HG_URL= -D COMPONENT_HG_REV= -D COMPONENT_NAME=import_profiler -D COMPONENT_FMRI=library/python/import-profiler -D COMPONENT_LICENSE_FILE=import_profiler.license -D TPNO= -D USERLAND_GIT_REMOTE=https://github.com/OpenIndiana/oi-userland.git -D USERLAND_GIT_BRANCH=python-import-profiler-0.0.3 -D USERLAND_GIT_REV=79d0f249c162a45e6ab7c5125322c32767b83aaa -D PYTHON_2.7_ONLY=# -D PYTHON_3.4_ONLY=# -D COMPONENT_SUMMARY=""Import profiler to find bottlenecks in import times."" -D COMPONENT_CLASSIFICATION="org.opensolaris.category.2008:Development/Python" -D COMPONENT_DESCRIPTION="" -D COMPONENT_LICENSE="MIT" /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.p5m \
        /export/home/zeka/oi-userland/transforms/license-changes /export/home/zeka/oi-userland/transforms/variant-cleanup /export/home/zeka/oi-userland/transforms/autopyc /export/home/zeka/oi-userland/transforms/defaults /export/home/zeka/oi-userland/transforms/actuators /export/home/zeka/oi-userland/transforms/devel /export/home/zeka/oi-userland/transforms/docs /export/home/zeka/oi-userland/transforms/locale /export/home/zeka/oi-userland/transforms/python-3-soabi /export/home/zeka/oi-userland/transforms/libtool-drop /export/home/zeka/oi-userland/transforms/ignore-libs /export/home/zeka/oi-userland/transforms/publish-cleanup | \
        sed -e '/^$/d' -e '/^#.*$/d' | uniq >/export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.mogrified
pkgmogrify: Error processing input arguments: File not found: 'profiler'
/export/home/zeka/oi-userland/tools/userland-mangler -D /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386/mangled -d /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386/mangled -d /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386 -d /export/home/zeka/oi-userland/components/python/import-profiler/build -d /export/home/zeka/oi-userland/components/python/import-profiler -d import_profiler-0.0.3 -m /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.mogrified >/export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.mangled
/usr/bin/pkgdepend generate -m -d /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386/mangled -d /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386 -d /export/home/zeka/oi-userland/components/python/import-profiler/build -d /export/home/zeka/oi-userland/components/python/import-profiler -d import_profiler-0.0.3 /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.mangled >/export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.depend
/usr/bin/pkgdepend resolve  -m /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-27.depend /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-34.depend /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.depend
/usr/bin/touch /export/home/zeka/oi-userland/components/python/import-profiler/build/.resolved-i386
VALIDATING MANIFEST CONTENT: /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-27.depend.res /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-34.depend.res /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.depend.res
/usr/bin/env PYTHONPATH=/export/home/zeka/oi-userland/tools/python PROTO_PATH="/export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386/mangled /export/home/zeka/oi-userland/components/python/import-profiler/build/prototype/i386 /export/home/zeka/oi-userland/components/python/import-profiler/build /export/home/zeka/oi-userland/components/python/import-profiler import_profiler-0.0.3" \
        /usr/bin/pkglint  \
                -f /export/home/zeka/oi-userland/tools/pkglintrc /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-27.depend.res /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-34.depend.res /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.depend.res
Lint engine setup...
CRITICAL lint.manifest003         Manifest /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-27.depend.res does not declare fmri.
CRITICAL lint.manifest003         Manifest /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler-34.depend.res does not declare fmri.
CRITICAL lint.manifest003         Manifest /export/home/zeka/oi-userland/components/python/import-profiler/build/manifest-i386-import-profiler.depend.res does not declare fmri.
Error: Fatal error in manifest - exiting.
gmake: *** [/export/home/zeka/oi-userland/make-rules/ips.mk:419: /export/home/zeka/oi-userland/components/python/import-profiler/build/.linted-i386] Error 1
```
All 3 res files are empty... So, not sure what else to change?
Any idea?

Other problem I have is:
```
$ git branch
  bzr-2.7.0
  oi/hipster
* python-import-profiler-0.0.3

$ git pull --rebase origin oi/hipster
From https://github.com/OpenIndiana/oi-userland
 * branch                oi/hipster -> FETCH_HEAD
Current branch python-import-profiler-0.0.3 is up to date.

$ git push 3eka python-import-profiler-0.0.3
Username for 'https://github.com': 3eka
Password for 'https://3eka@github.com': 
To https://github.com/3eka/oi-userland.git
 ! [rejected]            python-import-profiler-0.0.3 -> python-import-profiler-0.0.3 (non-fast-forward)
error: failed to push some refs to 'https://github.com/3eka/oi-userland.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

What I am doing wrong here (wanted to push changed files I use)?

Thanks and Regards.

**Comment by pyhalov** [comment]
Please, show the p5m manifest which causes issues. But I suspect it's a known oi-userland issue - it wrongly escapes strings with spaces while handling PYV files, so just use "Import profiler to find bottlenecks in import times." in p5m file, not $(COMPONENT_SUMMARY). 
As for second issue, you either have to force-push changes or reset to original commit and apply changes in separate commit.

File: components/python/import-profiler/Makefile [hunk 3]
```diff
@@ -28,7 +28,7 @@ COMPONENT_FMRI=         library/python/import-profiler
 COMPONENT_CLASSIFICATION=Development/Python
 COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
 COMPONENT_LICENSE=       MIT
-COMPONENT_SUMMARY=      "Import profiler to find bottlenecks in import times."
+COMPONENT_SUMMARY=      "Import_profiler_to_find_bottlenecks_in_import_times."
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: I meant removing COMPONENT_SUMMARY from Makefile and adding it directly to manifest

Comment (3eka) [reply components/python/import-profiler/Makefile]: Hi @pyhalov 
I am not sure how to do it!

Looked around and found (for example) that COMPONENT_SUMMARY IS defined in Makefile for python/geoip and even with spaces (although w/o quotes)!
```
$ grep COMPONENT_SUMMARY ~/oi-userland/components/python/geoip/Makefile
COMPONENT_SUMMARY=      MaxMind GeoIP Legacy Database - Python API
```
Please advice.

Regards.

Comment (pyhalov) [reply components/python/import-profiler/Makefile]: Hi.
Remove COMPONENT_SUMMARY from Makefile and add directly to p5m file, like it's done, for example, at https://github.com/OpenIndiana/oi-userland/blob/3f4d054e2924db4c4e9bd68846dc4f8617d961e2/components/python/pycairo/pycairo-PYVER.p5m#L28 . 

geoip doesn't use versioned (PYV) manifests, so it's not affected.

File: components/python/import-profiler/manifests/sample-manifest.p5m [hunk 4]
```diff
@@ -14,7 +14,7 @@
 #
 
 set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
-set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=pkg.summary value="A basic python import profiler to find bottlenecks in import times."
```
Comment (pyhalov) [inline components/python/import-profiler/manifests/sample-manifest.p5m]: Sorry for this taking so long, but please, don't touch sample-manifest. It's there for the next one updating package could do gmake sample-manifest and look at git diff.

File: components/python/import-profiler/import-profiler.license [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Resources:
```
Comment (pyhalov) [inline components/python/import-profiler/import-profiler.license]: put here the text of license, either from the module or from one of the locations.

File: components/python/import-profiler/Makefile [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+COMPONENT_FETCH_USER_AGENT=wget
+
+COMPONENT_POST_INSTALL_ACTION= \
+  gfind $(PROTO_DIR)$(PYTHON_VENDOR_PACKAGES.32) -type f \
+    -exec gawk '/^\#!.*python/{print FILENAME}{nextfile}' {} + \
+  | xargs gsed -i -e '1 s;^.*$$;\#!$(PYTHON.32);'
+
+# setup.py needs README.md, but file is missing in archive?
+COMPONENT_PRE_BUILD_ACTION=cd $(SOURCE_DIR) && echo "A basic python import profiler to find bottlenecks in import times." > README.md
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: why not just touch it?

File: components/python/import-profiler/Makefile [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+PYTHON_VERSIONS=	2.7 3.4
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/setup.py.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+COMPONENT_FETCH_USER_AGENT=wget
+
+COMPONENT_POST_INSTALL_ACTION= \
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: We don't need files in $(PROTO_DIR)$(PYTHON_VENDOR_PACKAGES) to have shebang

File: components/python/import-profiler/manifests/sample-manifest.p5m [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 Predrag Zečević
+#
+
+set name=pkg.summary value="A basic python import profiler to find bottlenecks in import times."
```
Comment (pyhalov) [inline components/python/import-profiler/manifests/sample-manifest.p5m]: Once again, please, leave sample-manifest as it was generated.

File: components/python/import-profiler/import-profiler.license (hunk 5 above)
Comment (pyhalov) [reply components/python/import-profiler/import-profiler.license]: No need in links here.

File: components/python/import-profiler/import-profiler-PYVER.p5m [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2017 Predrag Zečević
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
```
Comment (pyhalov) [inline components/python/import-profiler/import-profiler-PYVER.p5m]: FMRI misses -$(PYV), this will likely break manifest generation logic.

Comment (3eka) [reply components/python/import-profiler/import-profiler-PYVER.p5m]: Hopefully fixed properly...

File: components/python/import-profiler/Makefile (hunk 1 above)
Comment (pyhalov) [reply components/python/import-profiler/Makefile]: @3eka remove this line

File: components/python/import-profiler/Makefile [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    sha256:8ca93c521d3360390ee18eefd2bf666704ba4d95d6bd9d683bb9b50c6d48f0b5
+COMPONENT_ARCHIVE_URL=	https://pypi.python.org/packages/1f/cf/6a60ab41279a8c9a75c6370ccd41449bcc09a39385dd6c282bd8c123053a/$(COMPONENT_ARCHIVE)
+COMPONENT_FMRI=         library/python/import-profiler
+COMPONENT_CLASSIFICATION=Development/Python
+COMPONENT_LICENSE_FILE= $(COMPONENT_NAME).license
+COMPONENT_LICENSE=       MIT
+
+PYTHON_VERSIONS=	2.7 3.4
+
+include $(WS_TOP)/make-rules/prep.mk
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: This should have been $(WS_MAKE_RULES)/prep.mk and so on - style fixes

File: components/python/import-profiler/Makefile [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+COMPONENT_PRE_BUILD_ACTION=cd $(SOURCE_DIR) && touch README.md
+
+# common targets
+build:		$(BUILD_32)
+
+install:	$(INSTALL_32)
+
+test:           $(NO_TEST)
+
+REQUIRED_PACKAGES += runtime/python-$(PYVER)
```
Comment (pyhalov) [inline components/python/import-profiler/Makefile]: This is not correct, you need both python-34 and -27 to build the component.
Change it to 
REQUIRED_PACKAGE += runtime/python-27
REQUIRED_PACKAGE += runtime/python-34

File: components/python/import-profiler/import-profiler.license [hunk 12]
```diff
@@ -0,0 +1,14 @@
+Copyright 2017 David Cournapeau
```
Comment (pyhalov) [inline components/python/import-profiler/import-profiler.license]: License file should have been named import_profiler.license, as it's searched as $(COMPONENT_NAME).license

**Review by pyhalov — CHANGES_REQUESTED** [review]

**Comment by 3eka** [comment]
Done.

**Review by pyhalov — APPROVED** [review]

**Pull request merged**